sample/src/main/kotlin/dev/inkremental/sample/utils.kt

```diff
@@ -0,0 +1,13 @@
+package dev.inkremental.sample
+
+import android.view.View
+import dev.inkremental.Inkremental
+import dev.inkremental.dsl.android.init
+import dev.inkremental.dsl.android.view.ViewScope
+
+inline fun <T : View> ViewScope.initWith(crossinline viewInit: T.() -> Unit) {
```

Comment:
Why `init` is not typed in the first place? We have type information in the core and can safely cast view for init, aren't we?

Reply:
No, I don't think so :( Maximum we can do here is create init function with generic scope, like 
`fun <T : ViewScope> T.init(action: (View) -> Unit) = attr(INIT_LITERAL, action)`
But this won't help us with casting to appropriate types in the client code(in the target init lambda function).

Reply:
We could generate init method for each scope instead of using generic method

Reply:
Oh yes, that we can do)

Reply:
Being unable to find a fast solution, I've offloaded this into #85 as a nice-to-have thing, so that we can move on with this PR. Thanks for the idea!